UI refreshment

soosr opened the pull request:
Fixes https://github.com/zkSNACKs/WalletWasabi/issues/12294
Fixes https://github.com/zkSNACKs/WalletWasabi/issues/12136
Fixes https://github.com/zkSNACKs/WalletWasabi/issues/10534
Fixes https://github.com/zkSNACKs/WalletWasabi/issues/10584
Fixes https://github.com/zkSNACKs/WalletWasabi/issues/11270
Fixes https://github.com/zkSNACKs/WalletWasabi/issues/9620
Fixes https://github.com/zkSNACKs/WalletWasabi/issues/12736
Fixes https://github.com/zkSNACKs/WalletWasabi/issues/12760
Fixes https://github.com/zkSNACKs/WalletWasabi/issues/12761
Closes https://github.com/zkSNACKs/WalletWasabi/pull/12763

(Note: The installer icons on Mac needs to be adjusted in the packager, will be done after this is merged)

![image](https://github.com/zkSNACKs/WalletWasabi/assets/16364053/fe50fac4-4f1b-4801-8418-ac7b3e732ed9)
![image](https://github.com/zkSNACKs/WalletWasabi/assets/16364053/e1cccfbe-4ef2-4066-821c-f0b3a6edfcdd)

![image](https://github.com/zkSNACKs/WalletWasabi/assets/16364053/2f24e89b-3d47-405f-8fcd-49c7d99dbe8e)
![image](https://github.com/zkSNACKs/WalletWasabi/assets/16364053/bbf01454-3176-46a7-80c9-bd8957f925b3)

MaxHillebrand commented:
Beautiful.

danwalmsley commented:
Note re: https://github.com/zkSNACKs/WalletWasabi/pull/12260/commits/78bebdfe154671d6a2b53636b6068ffce89d2c10

This commit is needed due to some issues in Avalonias binding system, specifically with OneWayToSource and Readonly properties (like `IsPointerOver` and `IsSelected`)

The issues causes the `OneWayToSourceBinding` to be replaced with a `TwoWay` binding, and then the View sets the Property on the ViewModel, it fires `PropertyChanged` event, the View tries to update the readonly property, and then Sets the ViewModel property yet again, at this point we are in an endless loop and stackoverflow occurs. 

Avalonias binding system is being refactored and that may resolve this issue.

soosr commented:
@wieslawsoltes in the history the amount field should be colored in the case if it is positive. Could you implement it?

soosr commented:
> > @wieslawsoltes in the history the amount field should be colored in the case if it is positive. Could you implement it?
> 
> @soosr What color I should use?

6E934A

soosr commented:
@wieslawsoltes need your help.

For some reason in the Wallet Coins dialog, there is a weird gap between the child items. It doesn't reproduce in history.
I think it might be related to the Dialog, because if put the wallet home screen into a dialog, I can repro the issue in the history too.
Could you take a look?

Change the background color too make it more noticeable:
![image](https://github.com/zkSNACKs/WalletWasabi/assets/16364053/856a1b9b-af14-4098-bd68-d85b5508578a)

soosr commented:
> @soosr The ExpressionKeyFrame for scale animation depends on Target.Opacity so with different time the scale was not 1.0 and it caused rounding errors. Pushed fix, please try if it fixed the issue.

Fixed! Thanks.

soosr commented:
TODO:

1. Flashing animation in History doesn't work.
2. Scrolling in History breaks the action button section.
   - Have an item that has SpeedUp or Cancel action button.
   - Scroll down, scroll up.
   - See bug.
   
@SuperJMN Please investigate and fix the first one (commit directly to this branch).
@wieslawsoltes Please try to find a workaround for the second issue.

SuperJMN commented:
> TODO:
> 
> 1. Flashing animation in History doesn't work.
> @SuperJMN Please investigate and fix the first one (commit directly to this branch). @wieslawsoltes Please try to find a workaround for the second issue.

Fixed. But I can't push to your repo. I've created a PR for it: https://github.com/soosr/WalletWasabi/pull/27

MaxHillebrand reviewed (changes requested):
tested 44cd5b7dc227e8d89459102d18bcd21fcfefce2c

the action button inside a dialog is a white text, should it be black like the Done button?
![2024-04-03-153006](https://github.com/zkSNACKs/WalletWasabi/assets/30683012/53936e3b-ff24-472b-8008-d9d091618dc5)

MarnixCroes reviewed (commented):
- Wallet Coins dialog can be black/empty: go to wallet coins -> select coin to send -> at preview tx screen -> click back at top-right -> wallet coins dialog is empty
- in history, the sorting icon gets slightly highlighted when clicking to open another dialog
- sorting on _Status_ both at history and wallet coins gives different result: I expect the opposite between arrow up and down, but the result is not opposite sorting
- when clicking on the sorted arrow, I expect it to undo the sorting
- _date_ sorting at wallet coins?
- ban icon should be on the left side? from script type
![image](https://github.com/zkSNACKs/WalletWasabi/assets/93143998/c43ed19d-855d-4aba-8354-f5b97d3dab57)

Kruwed commented:
-Will the fee be added to the outgoing amounts for payments in the history? https://github.com/zkSNACKs/WalletWasabi/issues/10584#issuecomment-1785562579 Coinjoins show the fees paid, so this would keep the behavior consistent.

-The selection boxes on the "Wallet Coins" screen are inconsistent if you are viewing a pocket with multiple coins using the same label:

![WalletWasabi Fluent Desktop_WFrokxbTW2](https://github.com/zkSNACKs/WalletWasabi/assets/118238055/02bde574-a45d-4806-bf6a-46afcdd17d9c)

-The description text on the wallet coins screen could be improved to emphasize that entire coins will be spent without creation of change

-I am no longer able to click on the green bar under privacy progress to display the UTXOs ring

-Hovering over the bar below the privacy progress shows a tooltip for "private funds" and "semi private funds" but there is no tooltip for non private funds

yahiheb reviewed (commented):
Tested this a bit, here is some feedback specially for the main view:

- As stated by Clement above the actionable buttons get too much attention and make it hard to focus on other information. 
- The Send, Receive, Buy Anything buttons look unnecessarily big. Maybe their icons should be removed.
- The lists are confusing without a header title for each column.
- The list looks empty. Maybe add some horizontal spacing between columns.
- Maybe decrease the character size of the date column to make the amount more visible.
- The green color of the semi private coins is not visible.
- The checkbox and tx details buttons on the right looks very weird. They are on the left side on most software.
- When minimized Wasabi's icon is barely visible.

soosr commented:
@MaxHillebrand 

> the action button inside a dialog is a white text, should it be black like the Done button? ![2024-04-03-153006]

No. In Setting or Wallet Settings all the switches and actionable items are Green, White. so it fits better among them.
Changing it to Black, Green would kinda break the view as those setting buttons would lean more toward the dialog button.
![image](https://github.com/zkSNACKs/WalletWasabi/assets/16364053/0980d998-dc10-4f48-8cfe-40fb9a9567ea)


> The light mode seems to have more shadows underneith the tiles than the dark mode.

It's on purpose. In the Dark theme, There are no shadows of Tiles or the main area to have a clean look. In light mode, we had to use to have a good contrast between those UI elements. Sadly what worked for the dark theme, looked bad for the light.
This is fine. People don't use the light one, we only have it for those who have some eye disorder and cannot see properly on Dark theme.

soosr commented:
@MarnixCroes 

> * Wallet Coins dialog can be black/empty: go to wallet coins -> select coin to send -> at preview tx screen -> click back at top-right -> wallet coins dialog is empty
> * sorting on _Status_ both at history and wallet coins gives different result: I expect the opposite between arrow up and down, but the result is not opposite sorting
> * _date_ sorting at wallet coins?
> * ban icon should be on the left side? from script type

Thanks, @SuperJMN is on them.

> * in history, the sorting icon gets slightly highlighted when clicking to open another dialog

Thanks, I will take a look.

> * when clicking on the sorted arrow, I expect it to undo the sorting

TDG doesn't support undoing sorting. Won't be implemented.

soosr commented:
@turbolay 

> The contrast between how vivid is the colour palette of the actionable buttons and the paleness of the rest of the colours is so huge that it makes it really hard and exhausting to focus on the information that has to be focused on (such as the amounts etc) because you always have to fight against the attraction of these static buttons, which is especially true when those buttons have a white text, as pointed out by Max.
> 
> It's too late now I believe, as everything in this PR is built that way, but `SystemAccentColor` would benefit a lot from having less yellow. If you check dark mode green actionable buttons of other software (such as Github's `Comment` button) they are never as vivid, which makes them look important while not attracting too much the eye).

The green is the branding green and the only identification mark of our branding.
On the other hand, I see your point. Will discuss it with the management and marketing team if it is ok to decrease it's vividness.
Thanks for the report.

soosr commented:
@yahiheb 

> * As stated by Clement above the actionable buttons get too much attention and make it hard to focus on other information.

Will see what can be done. https://github.com/zkSNACKs/WalletWasabi/pull/12260#issuecomment-2036557951

> * The Send, Receive, Buy Anything buttons look unnecessarily big. Maybe their icons should be removed.

Icons are very important, won't be removed.
Decreased the padding also found that the spacing in BuyAnything button was not consistent, fixed.

> * The lists are confusing without a header title for each column.

They are fine, Bitcoin.design also suggests using a very similar layout.

> * The list looks empty. Maybe add some horizontal spacing between columns.

I don't what you mean, please elaborate more.

> * Maybe decrease the character size of the date column to make the amount more visible.

The fontsize of the date cells are even now one size smaller than the default, making it smaller one more look really bad.
Making the Amount one size bigger also looks really bad.
As of not the Amount is well noticeable, we can improve it when Avalonia 11.s is released with the font features.

> * The green color of the semi private coins is not visible.

The colors there were already adjusted and approved by the management. Won't be changed.

> * The checkbox and tx details buttons on the right looks very weird. They are on the left side on most software.

In order to know if you need to tick that checkbox first you need to read the row. So you can process it from left to right and make a decision as your focus is already at the end of the row.
Once you finish with the selection, your cursor and your attention are on the right so you are very close to the Dialog's next button.

It's much better to have it on the right, won't be changed.

> * When minimized Wasabi's icon is barely visible.

Tried to obey the branding and, will discuss it with the marketing team and see what can we do.

soosr commented:
All review comments have been explained or fixed, please do another round of testing. If nothing comes up tomorrow morning I will hit the merge button.

Thanks everyone!

MarnixCroes reviewed (commented):
- at CJ details the `Fees` has a minus/negative amount, while at outgoing tx the `Fee` is a positive value?
- everytime I open tx details of outgoing tx, the block hash value flickers a bit
- on tx's of today and yesterday I prefer to have the time displayed, without needing to hover over
- and when having done multiple coinjoins on the same date, I prefer to see the time instead of 10x date x
- that the scrollbar is only visible on hover (at multiple places) is not the best UX imo
- at wallet coins, the AScore of some coins can be a bit cut off: open wallet coins -> scroll down -> scroll back up-> AScore cut off last bit

<details>
<summary>it crashed</summary>
<br>

not  sure how exactly. CJ ended and I clicked wallet settings in searchbar
```
Exception type: System.ApplicationException

Message: Exception has been thrown in unobserved ThrownExceptions

Stack Trace:    at WalletWasabi.Fluent.Desktop.WasabiAppExtensions.<>c__DisplayClass0_2.<RunAsGuiAsync>b__5() in WalletWasabi.Fluent.Desktop/Program.cs:line 164
   at System.Reactive.Concurrency.Scheduler.Invoke(Action action)
   at System.Reactive.Concurrency.Scheduler.<>c.<Schedule>b__74_0(IScheduler _, Action a)
   at Avalonia.ReactiveUI.AvaloniaScheduler.Schedule[TState](TState state, TimeSpan dueTime, Func`3 action)
   at System.Reactive.Concurrency.Scheduler.Schedule(IScheduler scheduler, Action action)
   at WalletWasabi.Fluent.Desktop.WasabiAppExtensions.<>c.<RunAsGuiAsync>b__0_1(Exception ex) in WalletWasabi.Fluent.Desktop/Program.cs:line 164
   at System.Reactive.AnonymousSafeObserver`1.OnNext(T value)
   at System.Reactive.ObserveOnObserverNew`1.DrainShortRunning(IScheduler recursiveScheduler)
   at Avalonia.ReactiveUI.AvaloniaScheduler.Schedule[TState](TState state, TimeSpan dueTime, Func`3 action)
   at System.Reactive.ObserveOnObserverNew`1.Schedule()
   at System.Reactive.Subjects.Subject`1.OnNext(T value)
   at ReactiveUI.ScheduledSubject`1.OnNext(T value) in /_/src/ReactiveUI/Scheduler/ScheduledSubject.cs:line 59
   at ReactiveUI.ReactiveCommand`2.Execute(TParam parameter) in /_/src/ReactiveUI/ReactiveCommand/ReactiveCommand.cs:line 739
   at ReactiveUI.ReactiveCommandBase`2.ICommandExecute(Object parameter) in /_/src/ReactiveUI/ReactiveCommand/ReactiveCommandBase.cs:line 245
   at ReactiveUI.ReactiveCommandBase`2.System.Windows.Input.ICommand.Execute(Object parameter) in /_/src/ReactiveUI/ReactiveCommand/ReactiveCommandBase.cs:line 135
   at Avalonia.Xaml.Interactions.Core.InvokeCommandAction.Execute(Object sender, Object parameter) in /_/src/Avalonia.Xaml.Interactions/Core/InvokeCommandAction.cs:line 141
   at Avalonia.Xaml.Interactivity.Interaction.ExecuteActions(Object sender, ActionCollection actions, Object parameter) in /_/src/Avalonia.Xaml.Interactivity/Interaction.cs:line 86
   at Avalonia.Xaml.Interactions.Custom.ButtonClickEventTriggerBehavior.AssociatedObject_OnClick(Object sender, RoutedEventArgs e) in /_/src/Avalonia.Xaml.Interactions.Custom/ButtonClickEventTriggerBehavior.cs:line 56
   at Avalonia.Interactivity.EventRoute.RaiseEventImpl(RoutedEventArgs e)
   at Avalonia.Interactivity.Interactive.RaiseEvent(RoutedEventArgs e)
   at Avalonia.Controls.Button.OnClick()
   at Avalonia.Controls.Button.OnPointerPressed(PointerPressedEventArgs e)
   at Avalonia.Reactive.LightweightObservableBase`1.PublishNext(T value)
   at Avalonia.Interactivity.EventRoute.RaiseEventImpl(RoutedEventArgs e)
   at Avalonia.Interactivity.Interactive.RaiseEvent(RoutedEventArgs e)
   at Avalonia.Input.MouseDevice.MouseDown(IMouseDevice device, UInt64 timestamp, IInputElement root, Point p, PointerPointProperties properties, KeyModifiers inputModifiers, IInputElement hitTest)
   at Avalonia.Input.MouseDevice.ProcessRawEvent(RawPointerEventArgs e)
   at Avalonia.X11.X11PlatformThreading.RunLoop(CancellationToken cancellationToken)
   at Avalonia.Threading.DispatcherFrame.Run(IControlledDispatcherImpl impl)
   at Avalonia.Threading.Dispatcher.PushFrame(DispatcherFrame frame)
   at Avalonia.Threading.Dispatcher.MainLoop(CancellationToken cancellationToken)
   at Avalonia.Controls.ApplicationLifetimes.ClassicDesktopStyleApplicationLifetime.Start(String[] args)
   at Avalonia.ClassicDesktopStyleApplicationLifetimeExtensions.StartWithClassicDesktopLifetime(AppBuilder builder, String[] args, Action`1 lifetimeBuilder)
   at WalletWasabi.Fluent.Desktop.WasabiAppExtensions.<>c__DisplayClass0_0.<RunAsGuiAsync>b__0() in WalletWasabi.Fluent.Desktop/Program.cs:line 195
   at WalletWasabi.Daemon.WasabiApplication.RunAsync(Func`1 afterStarting) in WalletWasabi.Daemon/WasabiAppBuilder.cs:line 78
   at WalletWasabi.Fluent.Desktop.WasabiAppExtensions.RunAsGuiAsync(WasabiApplication app) in WalletWasabi.Fluent.Desktop/Program.cs:line 152
   at WalletWasabi.Fluent.Desktop.Program.Main(String[] args) in WalletWasabi.Fluent.Desktop/Program.cs:line 63

Inner Exception: Exception type: System.NullReferenceException

Message: Object reference not set to an instance of an object.

Stack Trace:    at WalletWasabi.Fluent.Extensions.CommandExtension.ExecuteIfCan(ICommand command, Object commandParam) in WalletWasabi.Fluent/Extensions/CommandExtension.cs:line 9
   at WalletWasabi.Fluent.ViewModels.Wallets.WalletViewModel.<CreateSearchItems>b__90_1() in WalletWasabi.Fluent/ViewModels/Wallets/WalletViewModel.cs:line 229
   at ReactiveUI.ReactiveCommand.<>c__DisplayClass9_0.<CreateFromTask>b__0() in /_/src/ReactiveUI/ReactiveCommand/ReactiveCommand.cs:line 433
   at ReactiveUI.ReactiveCommand.<>c__DisplayClass5_0`1.<CreateFromObservable>b__0(Unit _) in /_/src/ReactiveUI/ReactiveCommand/ReactiveCommand.cs:line 304
   at ReactiveUI.ReactiveCommand`2.Execute(TParam parameter) in /_/src/ReactiveUI/ReactiveCommand/ReactiveCommand.cs:line 720

Inner Exception: 
```

</details>

Kruwed commented:
The color for incoming amounts is a dull color (reminds me of dead grass). It lacks contrast with the background, could it be more bold?

![image](https://github.com/zkSNACKs/WalletWasabi/assets/118238055/3ce5b228-7607-4839-8423-1b3021fcf8b1)

Here's some examples of other wallets/services that use green for positive amounts, which appear more vibrant -

LNMarkets:

![lnm](https://github.com/zkSNACKs/WalletWasabi/assets/118238055/af55bbf6-406a-4589-bc7b-c67439fa8ca0)

Phoenix:

![phoenix](https://github.com/zkSNACKs/WalletWasabi/assets/118238055/949c0256-8c8c-49d8-adf1-0d30fe55a355)

Blixt:

![blixt](https://github.com/zkSNACKs/WalletWasabi/assets/118238055/e4acecc3-ebe9-49e7-9ff5-db5bc22cc58c)

soosr commented:
> * at CJ details the `Fees` has a minus/negative amount, while at outgoing tx the `Fee` is a positive value?

Fixed https://github.com/zkSNACKs/WalletWasabi/pull/12260/commits/1d88af0090aa87b87b75a2347474f608169ab2e2, https://github.com/zkSNACKs/WalletWasabi/pull/12260/commits/454540a486bb53c946a2000c9105b669bbc862b1
 
> * everytime I open tx details of outgoing tx, the block hash value flickers a bit

Seems like Avalonia rendering issue, won't be fixed for now.

> * on tx's of today and yesterday I prefer to have the time displayed, without needing to hover over

The time most of the time is irrelevant, Bitcoin.design also does as we do. Won't be changed.

> * and when having done multiple coinjoins on the same date, I prefer to see the time instead of 10x date x

Fixed https://github.com/zkSNACKs/WalletWasabi/pull/12260/commits/7f48819a25f21b12819619c1e27197823048105c.

> * that the scrollbar is only visible on hover (at multiple places) is not the best UX IMO

In the modern world where people are scrolling with a mouse scroll, gesture on a touchscreen or touchpad hiding the scrollbar to reduce the cognitive load is the best UX. Won't be changed.

> * at wallet coins, the AScore of some coins can be a bit cut off: open wallet coins -> scroll down -> scroll back up-> AScore cut off last bit

It was the same scrolling issue that we had with the Action area, fixed with @wieslawsoltes's custom control. https://github.com/zkSNACKs/WalletWasabi/pull/12260/commits/8783a236d41d7063a87bf47a34d0fd0d9a9e11e2

> it crashed

Was not related to the PR, https://github.com/zkSNACKs/WalletWasabi/pull/12787 fixed it.

soosr commented:
> The color for incoming amounts is a dull color (reminds me of dead grass). It lacks contrast with the background, could it be more bold?
> 
> ![image](https://private-user-images.githubusercontent.com/118238055/320742399-3ce5b228-7607-4839-8423-1b3021fcf8b1.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTI2NDYxODgsIm5iZiI6MTcxMjY0NTg4OCwicGF0aCI6Ii8xMTgyMzgwNTUvMzIwNzQyMzk5LTNjZTViMjI4LTc2MDctNDgzOS04NDIzLTFiMzAyMWZjZjhiMS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNDA5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDQwOVQwNjU4MDhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1kNTRiOTA3YTc1NzdhM2RkZDhkZjQ0ZWZlNDFlNjVjZDg0Zjg2ZDRjNmRiN2NhNjBiZTE2MTQ5ZDAzY2Q4Njg5JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.6snvvt3t5CDdXgXIObjlvY3B282EyvAusFysqdn6Um0)

That color pick was on purpose, nothing can be more emphasized than the Function buttons (Send, Receive)

> What do you think about using the same shade of green as the one that appears in the bar for fully private coins?
> 
> ![image](https://private-user-images.githubusercontent.com/118238055/320745374-528c8140-d100-490c-ae7a-d4e1c6c3faaa.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTI2NDYxODgsIm5iZiI6MTcxMjY0NTg4OCwicGF0aCI6Ii8xMTgyMzgwNTUvMzIwNzQ1Mzc0LTUyOGM4MTQwLWQxMDAtNDkwYy1hZTdhLWQ0ZTFjNmMzZmFhYS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNDA5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDQwOVQwNjU4MDhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT01OGQwMjI2M2NlNjc2MmQ3YWQ4N2UzYWNlMjJiNzk1ZTNjMDFiNDRmOTczZDZkNTA4MjFlNGZkMjg1YWVmYzNkJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.Pm0K8M0vJPIQ3Ym0KC3ClUJFtMac2jytB34Dkg-O2Lg)

That green is dedicated to indicating if something is Private. (Semi-private also has its own color) So users can pair that color with privacy. Should not be used for anything else.

##

The current color is good enough, won't be changed.

MaxHillebrand commented:
Congrats, this is a huge PR. 🚀

molnard commented:
![RolandTannerWolfOfWallStreetGIF](https://github.com/zkSNACKs/WalletWasabi/assets/9844978/dc992c72-df41-48fa-b8cf-68ea5bd85219)
